packages/components/src/modal/style.scss

```diff
@@ -68,9 +68,13 @@
 	background: $white;
 	align-items: center;
 	height: $header-height;
+	z-index: z-index(".components-modal__header");
+	// For z-index to take effect, the element must be positioned. A "sticky"
+	// element is positioned, but since this is not supported in IE11,
+	// "relative" is used as a fallback.
+	position: relative;
```

Comment:
As an FYI if folks need a stacking context, any of the other rules on https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context will also do the trick, in case changing position has other side effects folks will avoid.

https://developer.mozilla.org/en-US/docs/Web/CSS/isolation for example only does this, but it's unsupported by IE.